docs/release-process.md

```diff
@@ -0,0 +1,225 @@
+This guide illustrates how to perform a release for Apache pulsar-client-go.
+
+In general, you need to perform the following steps:
+
+1. Create a release branch.
+2. Update the version and tag of a package.
+3. Build and inspect an artifact.
+4. Sign and stage the artifacts.
+5. Write a release note.
+6. Run a vote.
+7. Promote the release.
+8. Update the release note.
+9. Announce the release.
+
+## Steps in detail
+
+1. Create a release branch.
+
+This example creates a branch `branch-0.X.0` based on `master` where tags will be generated and new fixes will be applied as parts of the maintenance for the release. `0.X.0` is the version of the release.
+
+The branch needs only to be created when creating a minor release rather than a patch release.
+
+Eg: When creating `0.1.0` release, will be creating the branch `branch-0.1.0`, but for `0.1.2` we would keep using the old `branch-0.1.0`.
+
+In these instructions, I'm referring to an fictitious release `0.X.0`. Change the release version in the examples accordingly with the real version.
+
+It is recommended to create a fresh clone of the repository to avoid any local files interfering in the process:
+
+```
+git clone git@github.com:apache/pulsar-client-go.git
+cd pulsar-client-go
+git checkout -b branch-0.X.0 origin/master
+```
+
+2. Update the version and tag of a package.
+
+Update the information of the new release to the `VERSION` file and `stable.txt` file and send a PR for requesting the changes.
+
+During the release process, you can create a "candidate" tag which will get promoted to the "real" final tag after verification and approval.
+
+```
+# Commit 
+$ git commit -m "Release 0.X.0" -a
+
+# Create a "candidate" tag
+$ git tag -u $USER@apache.org v0.X.0-candidate-1 -m 'Release v0.X.0-candidate-1'
+
+# Push both the branch and the tag to Github repo
+git push origin branch-0.X.0
+git push origin v0.X.0-candidate-1
+```
+
+3. Build and inspect an artifact.
+
+Generate a release candidate package.
+
+```
+$ tar -zcvf apache-pulsar-client-go-0.X.0-src.tar.gz .
+```
+
+If you have not installed Go, install it according to the [installation instruction](http://golang.org/doc/install).
+
+> NOTE: Since the `go mod` package management tool is used in this project, **Go 1.11 or higher** version is required.
+
+After the build, inspect the artifacts:
+
+Run the following command to verify the license headers in the source files:
+
+```
+go test license_test.go
+```
+
+Run the standalone Pulsar service and check that the Go client can produce and consume messages correctly:
+
+```
+$ cd examples/consumer
```

Comment:
Same here, we're running a more extensive set of tests for each commit. I don't think these manual steps will be useful in catching potential problems.